### ISSUE TITLE
Need a beta repo?

### ISSUE THREAD

**Issue by tknorris** [Razzeee/script.trakt#125]
Do you need a Kodi addon repo to push to for testing your changes to this addon? If so, I'm happy to start pulling your dev branch into my beta repo hosted here on github so that people can just install my beta repo in Kodi to do testing.
If not, that's cool too. 
Just wanted to offer in case it would make things easier to get your changes tested and deployed.

**Comment by kodispot** [comment]
+1

**Comment by Razzeee** [comment]
Would be nice to have that tested before we push to the kodi repo. And would be great to minimize my fear, as this is the first time I release something to the official repo.

**Comment by freeman212** [comment]
cool :)

**Comment by kodispot** [comment]
And I think you have to start to change the version number to let the repo push the updates.

**Comment by Razzeee** [comment]
@kodispot that's not gonna happen until I push to the official kodi repo

**Comment by freeman212** [comment]
it would be required to get updates from the beta repo too

**Comment by kodispot** [comment]
@Razzeee You need to do that if you want a repo that push your updates automatically..

**Comment by tknorris** [comment]
It's OK. I can do a local branch that bumps the version number each time I do a pull and copy that into the repo. I can mark them 3.0.0~betaX. XBMC will treat 3.0.0 as newer than any of those.

**Comment by tknorris** [comment]
Let me know if you are ready and I'll do the first pull today.

**Comment by Razzeee** [comment]
@tknorris just pull it. Should be working fine. The only thing thats not ready for the real 3.0.0 release are the translations. I'm just waiting for a new transiflex sync...

**Comment by tknorris** [comment]
OK, I just pulled it into my beta repo, bumped the version number to 3.0.0~beta1, and tested on one of my machines. I should be able to just keep rebasing off your updates on your dev branch as you go and bumping versions each time it needs to be pushed to the repo again.

It should be no issue for me to update within a few hours of you needing to do another pull, so just comment in this issue any time you want me to pull again. My beta repo zip is here: https://github.com/tknorris/tknorris-beta-repo/blob/master/zips/repository.tknorris.beta/repository.tknorris.beta-1.0.1.zip?raw=true if you want to test yourself.

**Comment by s7eele** [comment]
great news, especially since i already use your beta repo @tknorris, this should get the trakt add-on to a larger test base as well

im curious how this is going to affect testing though, im wondering if we will now see people reporting on both tvaddons and kodi forums as well as github, also wondering if this will affect which repository it ultimately gets assigned to, will the main thread remain on kodi forum or will it migrate to tvaddons

just thinking out loud, i realize these arent immediate concerns but wrt reporting issues it seems it might get a bit confusing, this obviously becomes less of an issue as the add-on nears production release

**Comment by tknorris** [comment]
I don't think it would make any sense to make the official repo for the trakt addon to be tvaddons. The official kodi repo is something everyone has and there's nothing questionable in the addon that would potentially prevent it from being in the official repo which is the main thing that the tvaddons repo provides.

I don't forsee any testing issues because all the ones in my beta repo will be marked with a beta version that the user can clearly see which should clearly differentiate them from direct github pulls or builds from the official kodi repository.

As far as that playlist-sync branch goes, there still shouldn't be any conflict because the version number in razzeee's repo are marked as 3.0.0 while the ones in my beta repo are marked as 3.0.0~betaX which means XBMC treats razzeee's versions as later versions that the one in my beta repo.

Note, I don't think which repo the addon is in should have any affect on where it's discussed/supported from. It's clear to me at least that this isn't a tvaddons addon, my repo is just a convient place to put it to be tested.

**Comment by s7eele** [comment]
completely understand your points and you know i meant no offense @tknorris, i often look to you for guidance on these issues so as always take what i say with a grain of salt (or SALTS lol?)

imho confusion has been a significant issue wrt getting people to the correct places to download and report issues, or more importantly where to look for existing issues before submitting new ones, central management may not be required but certainly helps, all based on my experience crossing forums (trakt/kodi/tvaddons) and github, thats all ive got to go on, your the expert wrt repositories afaic

its all quite clear to me as well, even counting the fact i aint so smart but i think a large majority will get quite confused and then get even more confused when production is released via a different repository, i also want to note that i agree in principle that an add-on should be able to be discussed anywhere but there are rules/regulations that say differently, for good reason in many cases, we ran into a couple issues already on kodi forum when it came to discussing .strm file issues and the add-ons most use to create them

shutting up now since i was never really asked for an opinion or to comment on the matter, apologies to all

**Comment by tknorris** [comment]
You shouldn't have to disable anything. Since the one in my repo is 3.0.0~beta1 xbmc will treat it as an older version that anything you install directly from razzee's github since he has it listed it as 3.0.0. I used that version number scheme purposely so that when the 3.0.0 final build actually gets released it will automatically suplant any build that someone might have installed from my beta repo.

**Comment by ashtonaut** [comment]
Fully support the idea of adding to the tknorris beta repo. Will make installing and testing much easier. Lack of a repo for the beta builds has prevented me testing as much as I would ordinarily like to.

**Comment by s7eele** [comment]
they were just my thoughts based on what ive seen across the forums, i am quite comfortable using either method even if its just for a couple weeks

@tknorris - was worried about the other way around wrt disabling your repo, looked kinda funny seeing both versions enabled when looking through your repository, shouldnt be a problem but if i have any questions you know ill ask (a weak lol?)

**Comment by tknorris** [comment]
I think you are referring to have a pure-github version installed and the beta one listed from the repo? If so, the issue there is XBMC treats them as two different addons because the addon path when you install from github (incorrectly) includes the github branch you pulled the zip from. Doing that should be avoided as you'll end up with two versions of the service running at the same time which will double scrobble, etc. This is the very reason I mentioned having a test repo because allowing people to install directly from github introduces a buch of extra variables that are hard to account for.

On the other hand, if you mean seeing both the kodi repo version as enabled and also the one in my beta repo enabled when 3.0.0-beta1 is installed, that's normal xbmc behaviour as it recognizes when you have a more recent version of an addon installed and shows them both as enabled under the assumption that the more recent version replaces the older one.

**Comment by s7eele** [comment]
no no no, i was referring to the second case and i actually understand all of it, lol

i manually install to the 'base' directory (script.trakt) when pulling from github for the reasons you mentioned

youve again proved why i consider you the expert in this matter, thx @tknorris

**Comment by RavRob** [comment]
Just a question on this.  If I installe from @tknorris beta repo, as new pulls (beta 2, 3 etc...) happen the addon will update itself as it should.  My question is this:  once the final release is pulled to Kodi's repo, will it over-write the beta from tknorris or will I have two?

**Comment by Razzeee** [comment]
Keep in mind, the dev branch might break again.

**Comment by tknorris** [comment]
Yes, as long as I keep releasing to my beta repo as 3.0.0~betaX, and when the official release is done, it's released as 3.0.0, it will overwrite what's in my beta repo. As Razzeee pointed out though, it's possible that the beta will break again in the future. That why it's a beta...;-)

**Comment by RavRob** [comment]
Understood.  Tx

**Comment by tknorris** [comment]
Razzeee, feel free to let me know whenver you want me to pull again to do another repo build. I'll do it as often or as little as you like. If I don't hear from you, I'll keep an eye on the dev branch and pull when it looks like there are enough changes to warrant a build. It's literally only two scripts I have to run to do a build so it takes very little time on my part.

**Comment by Razzeee** [comment]
Got it, can you do one now? 
Next one probably tomorrow when the playback branch is merged

**Comment by tknorris** [comment]
Done.

**Comment by Razzeee** [comment]
@tknorris 
here we go, another one please!

**Comment by tknorris** [comment]
I added a comment on that question about the math calc. Do you want me to wait to do the build until you have time to look at that?

**Comment by Razzeee** [comment]
No, no need, your building from the dev version and that one is about the playback-sync branch

**Comment by tknorris** [comment]
beta3 up

**Comment by Razzeee** [comment]
Thanks!
I'll close the issue we can still comment here, just so it doesn't show up in the open issues.

**Comment by Razzeee** [comment]
@tknorris another would be great

**Comment by tknorris** [comment]
I'll push one in a within an hour.

**Comment by Razzeee** [comment]
Easy no need to rush

**Comment by s7eele** [comment]
dont think @tknorris knows any other speed, hes super fast responding to support requests on tvaddons forum and the few issues found with his add-ons are generally fixed just as fast (often in a matter of hours)

**Comment by tknorris** [comment]
done.

**Comment by s7eele** [comment]
see lol

**Comment by freeman212** [comment]
haha you guys :joy:

**Comment by Razzeee** [comment]
@tknorris would be nice if you could do another sync

**Comment by tknorris** [comment]
done. I'm so scared to see if I get the movie sync issue again ;-)

**Comment by Razzeee** [comment]
@tknorris me too

**Comment by RavRob** [comment]
Just a comment.  With the latest Beta release (B5) It failed to scrobble "Gangland Undercover" until I rebooted Kodi.  It was scrobbling other shows though, just this one.  A reboot fixed that and it scrobbled well after that.  I don't think it is an addon problem, rather a problem at my end but in case others have similar issues.  Reboot....

**Comment by Razzeee** [comment]
@tknorris can you do a new version?

**Comment by tknorris** [comment]
done. 3.0.0~beta6 is pushed.

**Comment by kodispot** [comment]
@Razzeee Could You Post Here Some Kind Of Changelog When You Request An Update To tknorris ? ...

**Comment by Razzeee** [comment]
Sorry, but you will have to read the commit log instead. I would just recite myself.

**Comment by tknorris** [comment]
IF you check the commit log of my forked repo you can see the commits between builds: https://github.com/tknorris/script.trakt/commits/dev

**Comment by Razzeee** [comment]
@tknorris 
can you do another one? just to be sure the version change of the dependencies doesn't break anything?

**Comment by tknorris** [comment]
done. 3.0.0~beta7 pushed to repo

**Comment by RavRob** [comment]
Beta 7 working fine.  Nothing broken

**Comment by Razzeee** [comment]
@tknorris 
new beta would be really appreciated

**Comment by tknorris** [comment]
done. 3.0.0~beta8 pushed to repo

**Comment by Razzeee** [comment]
@tknorris 
New build please :)

**Comment by tknorris** [comment]
done. 3.0.0~beta9 pushed to repo

**Comment by Razzeee** [comment]
@tknorris 
can you do another one? but one important thing can you add these to your repo before you update the beta?
https://github.com/Razzeee/script.module.arrow
https://github.com/Razzeee/script.module.dateutil
https://github.com/Razzeee/script.module.trakt

**Comment by tknorris** [comment]
Are those new repos pretty stable or do I need to work them as well to plan for updates? I'd rather not clutter up my workspace with all those projects if they aren't needed.

**Comment by Razzeee** [comment]
@tknorris 
I don't think I will need to update them. 
It's working here, with the changes, I just want to be sure the dependencys are getting resolved correctly. I'm expecting a official repo push this week, so it won't be needed for long on your repo.
But since they are the first modules I ever packed for Kodi I'm kinda nervous.

**Comment by tknorris** [comment]
OK, I pushed the dependent zips in one commit and then another to bump script.trakt to 3.0.0~beta10. I think I got everything right. All the deletes make things a little more challenging so let me know if something is wrong.

**Comment by tknorris** [comment]
So, the beta10 is throwing an unmet dependency error. I'm not exactly sure which one is the problem, but I think its that the version number you chose is 2.1.1-beta, but it's supposed to be 2.1.1~beta. (i.e. a "tilde" not a "dash" before beta). trakt.py installs just fine on it's own and installs the dateutil and arrow dependencies when it does, but script.trakt doesn't seem to see trakt.py. I'll try changing the version numbers in my repo and if I fix it, you can backport the numbers to your repos.

**Comment by tknorris** [comment]
OK, I found the problem. There's a type in script.trakt's addon.xml referring to "script.module.dateutl". Instead it reads "script.module.datutil"

**Comment by Razzeee** [comment]
at all using this beta repo please let me know if you got problems with the new version.

I'm also looking for some people ( @s7eele @RavRob @freeman212 ?) to test the update comming from 2.7.2 or prior, would be really great as I need to work for 8 hours and could be back at it in like 10 hours, so it's likely someone else can test it before me.

1. Remove all traces of the trakt addon
2. deinstall the beta repo
3. Install trakt from the officail kodi repo (the old version)
4. Set it up in the settings, maybe do a reboot
5. Add the beta repo again, at this point the trakt addon should auto update to 3.0.0~X
6. See if it is working okay
7. Let me know what happend

@tknorris thank you so much. this is indeed why we test, really glad I didn't push it without this...

**Comment by s7eele** [comment]
@razze need to fit a couple hours sleep in but i can definitely run through it in a bit, i havent been using the repo, ive been doing manually still, speaking of which i just left a post on kodi forum that you should be notified about, involves a problem manually installing most recent dev manually (dateutil, arrow, trakt), may not exist long enough to matter :/

**Comment by Razzeee** [comment]
@tknorris 
sorry again, I changed script.module.trakt version number to the correct one (with tilde)

Would be really really great if you could bump script.module.trakt and this repo again :/
Not sure if I need to increase both versions for that fix

**Comment by Razzeee** [comment]
@s7eele sure 2.3.2 anything prior to 3.0.0 should yield a valid test
and just pick the system your most comfortable with to test ;)

**Comment by s7eele** [comment]
understood, i was thinking more about splitting between us testers, all good

**Comment by s7eele** [comment]
@razzeee should i remove dateutil, arrow or trakt first, i installed them manually

**Comment by Razzeee** [comment]
that would be great, maybe even remove six and requests

**Comment by s7eele** [comment]
alright, i know your ready to get this pushed so sorry but its gonna be four hours or so, something just came up, ill get to it sooner if possible, ill check back before starting so i dont waste time if somethings already been done

**Comment by Razzeee** [comment]
Easy, I'm stuck at work, so if it will take till tonight, thats okay for me.

**Comment by freeman212** [comment]
same, I'm at work too. still 6 hours to go.. otherwise I would like to help.

**Comment by RavRob** [comment]
Sorry guys. My day is just starting here. It will be at least 10 hours before I can get to testing as requested. I am on my way to work now and won't be back for a while.

**Comment by ShedDweller** [comment]
Hey all... Firstly THANK YOU. I've been lurking and following your progress on here with great interest. so seeing as you were all busy though i'd report your success, so you knew your efforts were worthwhile!

New Kodi 4.2beta install on Android 4.4.4. Fresh install and configuration of the original trakt addon from the Kodi default repo (rebooted) and then the beta repo which installed the necessary extras (reboot) 

It updated everything perfectly. Local library .mp4 files (and .strm from genesis) added to collection on trakt.tv.  All watched movies and episodes from trakt.tv updated to Kodi. Added some new local content and mp4 and strm and these updated the track.tv site straight away. The % progress "you are currently watching" showed up on the website too!

Fantastic work!
Thanks again.

**Comment by Razzeee** [comment]
@ShedDweller
thank you for reporting! Really appreciated. 
I'll test some more and will send it to the official repo later tonight.

**Comment by tknorris** [comment]
3.0.0~beta11 pushed with latest dev branch changes.

**Comment by Razzeee** [comment]
@tknorris
could you do one last one, I just fixed the ratings

**Comment by tknorris** [comment]
3.0.0~beta12 pushed

**Comment by s7eele** [comment]
@ShedDweller if you dont mind i have a quick question, just so i can keep things straight in my head:

what hardware are you on?

i realize it probably wont come up but jic, thx for your help

nice to meet u btw

**Comment by ShedDweller** [comment]
Np @s7eele happy to help... 

That test mentioned was just on my phone... A OnePlus One. 

I've since also added it to a Kodi 14.1 install on a Windows 10 zotac ci320 which also seems fine syncing and scrobbeling, I've not had any errors anyway! I've not tested everything though... 

Any other questions just shout.

**Comment by s7eele** [comment]
@razzeee @tknorris i couldnt get to a windows machine so i ran through it on my ouya (android again), upgrade went perfectly and everything appears to be working well AFTER restarting kodi but before restart there were issues, detail follows:

i followed steps making sure to remove ALL traces (add-ons, modules, packages, cache, etc.), the only thing i didnt remove was the requests module, it was being used by several other add-ons, i cleared directories for everything i removed, i shut down and restarted kodi in between the installation of 2.3.1and installion of the beta repository, i did not use autoupdate, i manually updated after receiving the notification that the update was available, not sure if any of that applies but after succesfully updating from 2.3.1 to 3.0.0 i encountered several issues UNTIL I RESTARTED kodi, before restarting kodi scrobbling and watched status worked (made it to trakt) BUT i did not see the notifications (add-on was configured to show them), the rating dialogue box showed up with hearts visible but no text and i couldnt select anything, i was forced to press the back button to get rid of rating box, a manual sync showed similar issues but it may have worked, the only thing visible in the dialogue besides the progress bar was '- x%', the percentage and progress bar worked as normal (progressing through completion) but nothing else was visible, i captured logs if necessary but as mentioned at the beginning A RESTART OF KODI CLEARED THINGS UP and everything now appears to be working fine, not sure if simply telling people to restart is acceptable, for that matter it could have been an isolated issue

i will test windows in just a bit and see if the same thing happens there, i also thought id try the upgrade process with version 2.3.2 available on github, i feel pretty sure some out there are using it still

sorry for length of comment but had to get the details out, be back a bit later, wife will be home shortly so gotta talk to her and stuff LOL

**Comment by Razzeee** [comment]
I can live with telling the people to restart. This is not a showstopper for me.

**Comment by RavRob** [comment]
I also had to restart. No Problems biggie. I always restart anyway just
because
#1 machine
 Windows 7
Helix 14.1

#2 machine
Windows 7
Isengad 15 alpha 1


On Thursday, March 5, 2015, Razzeee <notifications@github.com> wrote:

> I can live with telling the people to restart. This is not a showstopper
> for me.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Razzeee/script.trakt/issues/125#issuecomment-77473993>
> .
>


-- 
Sent from Gmail Mobile

**Comment by s7eele** [comment]
definitely not a problem for me, just being thorough as usual, thought you guys might recognize the behavior or know a quick fix, you are magic right? lolol

**Comment by Razzeee** [comment]
Probably won't.
But its pretty nice to know what will happen when you don't restart, as there will probably be some questions.

**Comment by s7eele** [comment]
a note in readme and initial post might be in order @razzeee

**Comment by tknorris** [comment]
FYI, it looks like the dependent modules are now in the main repo so I had to remove them from my beta repo as the zips had a different md5 than the ones in my repo and were causing a conflict. If we add them back in the future we need to give them new versions numbers.

**Comment by s7eele** [comment]
i was just wondering about that situation @tknorris, reviewing things after installation of the add-on via kodi repo got me to thinking about how it would all interact (both repos enabled), i wasnt sure so i decided to disable your beta repo and do some research, not sure what the plan is from here on out but your beta repo is one of the few i use and i dont wanna leave it disabled so wanted to ask:

will having both enabled cause any problems?

i think we talked about this with regard to your official repo but i cant recall it completely and it may not apply in this case anyway, thx

**Comment by tknorris** [comment]
Now that I deteled the extra modules, everything should be fine. 3.0.0 will override the 3.0.0~beta versions in my repo.

**Comment by s7eele** [comment]
most excellent  👍

**Comment by tknorris** [comment]
3.0.1~beta1 of script.trakt and 2.1.3~beta1 of script.module.trakt pushed to my beta repo.

**Comment by tknorris** [comment]
3.0.1~beta2 pushed.

**Comment by Razzeee** [comment]
Well, we cound go for another beta, contextMenu and toggle watched should be working.
Not really proud of the implementation I did for the shows part of toggle watched but, well it seems like its working.

**Comment by RavRob** [comment]
So what is this "toggle watched" supposed to do?  Mark as unwatched from the Kodi library?

**Comment by Razzeee** [comment]
Other way around. Mark as watched from the Kodi library.

**Comment by tknorris** [comment]
Isn't what it's supposed to do is to changed the watched status of the given media on trakt?

**Comment by Razzeee** [comment]
Yeah, but I don't think it's working on already watched items of you want to unwatch them. So only the other way around.

**Comment by tknorris** [comment]
OK, do you want me to go ahead and do the PEP8 cleanup before this beta?

**Comment by Razzeee** [comment]
Sure

**Comment by tknorris** [comment]
3.0.1~beta3 pushed

**Comment by s7eele** [comment]
awesome work as usual guys

**Comment by s7eele** [comment]
now im just gonna have to ask because im not seeing it through beta in tk's repository either, how do i pull up the context menu @razzeee or anyone

**Comment by tknorris** [comment]
I assume you have to have something that can invoke that function in the script. I'm not aware of any addon that has that function. You could bind a key to the command shown here: https://github.com/trakt/script.trakt/wiki/Action%3A-contextmenu

**Comment by s7eele** [comment]
ok, i saw the command line of it somewhere along the way but for some reason i was thinking this would actually display the context menu, this basically makes it available to be called? kinda like the way you calll rating from salts

**Comment by tknorris** [comment]
yes.

**Comment by s7eele** [comment]
thx, i can work with that, curl example u gave me will probably help too ;-)

**Comment by Razzeee** [comment]
You can bind it to a button like in this example https://github.com/trakt/script.trakt/wiki/Action:-togglewatched that's probably what your looking for

**Comment by s7eele** [comment]
yeah, my bad, i saw the wiki updates and the examples but for whatever reason i was still expecting to 'see' something, misunderstanding, sorry

**Comment by Razzeee** [comment]
Can you do a new beta? Please also do a new one for module.trakt

**Comment by tknorris** [comment]
Should be done. 2.2.0 of script.module.trakt pushed and 3.0.1~beta4 of script.trakt pushed.

**Comment by Razzeee** [comment]
To everybody using this, please report back with anything, good or bad. I would like to push a release, as the next step for this addon will be disbanding token auth and changing it to pin auth.

**Comment by Razzeee** [comment]
@tknorris 
can you please push another beta?
and also update
https://github.com/Razzeee/script.module.dateutil
and
https://github.com/Razzeee/script.module.trakt

**Comment by tknorris** [comment]
done. 2.3.0 of module.trakt; 3.0.1~beta8 of script.trakt; and 2.4.2 of dateutil all pushed to my beta repo

**Comment by Razzeee** [comment]
Can you do another quick one for the main script?

**Comment by tknorris** [comment]
3.0.1~beta9 of script.trakt pushed.

**Comment by Razzeee** [comment]
@tknorris 
sorry to bother you again.
can you push an update to the trakt module and the main script?
Thank you

**Comment by tknorris** [comment]
No problem. It's very little work for me to do a push to the repo, most of it is automated.

I just do a git pull; then run my copy script to copy the addon source into the repo, and then a generator script to create the zips and update the repo metadata. Start to finish it's less than 5 minutes.

**Comment by Razzeee** [comment]
Anyone having any problems with the latest beta?

**Comment by tknorris** [comment]
I'm not. No more duplicate syncs.

**Comment by RavRob** [comment]
Working like a charm here also.   No dupes and no "wrong pin" notifications.

**Comment by s7eele** [comment]
all well here

**Comment by Razzeee** [comment]
Good to hear, I'll try to test if an upgrade from the old one resets the login.

**Comment by Razzeee** [comment]
Can you do a new build @tknorris ?
Dev branch + new version of https://github.com/Razzeee/script.module.trakt

**Comment by tknorris** [comment]
Done and done. 2.3.2 of script.module.trakt pushed and 3.0.2~beta5 of script.trakt pushed.

The new pin dialog looks great. One thing you might want to check is other popular skins. I had issues with the original dialog not looking good on some popular skins due to font issues. If the font you use isn't defined in that skin, Kodi will fall back to it's default font which is can be bigger than what you want.

**Comment by Razzeee** [comment]
Anyone having problems with the new build?

@tknorris 
Not sure but knowing @vonH he probably has already checked this

**Comment by RavRob** [comment]
Seems all good at my end
Windows 7
Isenguard Beta 2

**Comment by vonH** [comment]
There's not much we can do about fonts. I will see about adding a couple of options to the defaults, purely because it's getting annoying.

@Razzeee :)

**Comment by Razzeee** [comment]
@vonH Yeah I know. Still waiting for some development on that. I really want my screensaver with clock etc :(

**Comment by vonH** [comment]
@Razzeee There is actually an easy solution. But it's probably not going to happen. I fully agree with you, however.

**Comment by Razzeee** [comment]
Last call, planning to send it out today.

**Comment by tknorris** [comment]
No issues here.

**Comment by vonH** [comment]
@Razzeee works fine

**Comment by RavRob** [comment]
All is well here as well

**Comment by s7eele** [comment]
all good here

**Comment by tknorris** [comment]
Seems like issues/code changes have slowed down alot. Is it a good time to push a beta?

**Comment by Razzeee** [comment]
Sure, my attention is shifted at the moment.

**Comment by tknorris** [comment]
3.0.4~beta1 pushed

**Comment by Razzeee** [comment]
Can we get a new beta? Last changes for 3.0.4 are in and I would like this to be the last gotham release.

**Comment by tknorris** [comment]
Sure, a couple of small comments on this commit though: https://github.com/Razzeee/script.trakt/commit/c6363be8fff34ec4128a759c39fdc7d41d056486

In python you can actually do 1970 < date < 2038, rather than the "or" logic.
Also, there is a way you can get pre-1970 dates using datetime I have that code here: https://github.com/tknorris/salts/blob/master/salts_lib/utils.py#L269

Let me know if you want me to go ahead w/ the beta or you want to make some changes to that code.

**Comment by Razzeee** [comment]
```1970 < date < 2038```
would be a between I'm using the opposite case and I wasn't sure how to invert it not sure if ```not``` would have worked.

Would be nice to get every date to work but mostly these are errors in kodi, trakt or on the users end. And I don't really like the ```iso_2_utc``` as it's not nice to understand (as everything with time and timezones) and it's not 100% correct, from what I understand from a quick look?

**Comment by tknorris** [comment]
not (1970 <= date <= 2038) should work. I did some quick testing, and it seemed to return the correct results.

I think iso_2_utc is 100% correct, but if it's not, I'd like to know the ways in which it isn't.

However, my main point was if you already have a negative epoch value, you can still get the correct datetime value by using the timedelta() trick rather than calling localtime():

    datetime.datetime.fromtimestamp(0) + datetime.timedelta(seconds=utc_air_time)

**Comment by Razzeee** [comment]
just figured because of the "close enough" comment on the one part ;)

**Comment by tknorris** [comment]
That refers to the fact that total_seconds() returns a floating point for seconds, while that 2nd calculation only returns an int. So, that means total_seconds() returns the fractional seconds while my other calculation does not. That's what I mean by "close enough" ;-)

**Comment by Razzeee** [comment]
Well, I would like to get a beta and a final version for gotham out there.
I think your code will work for this part, but as it's one of the key pieces and I can't just copy and paste it, it will have to wait for the next one.

**Comment by tknorris** [comment]
Works for me. I'll push one as soon as I get a chance.

**Comment by tknorris** [comment]
3.0.4~beta2 pushed

**Comment by Razzeee** [comment]
Can you push a new one?
Thank you!

**Comment by tknorris** [comment]
done

**Comment by Razzeee** [comment]
Anyone having problems with the beta?

**Comment by RavRob** [comment]
No issues so far on Windows 8.1 running Isenguard 15.1 or on Windows 10
running Jarvis Alpha 1

**Comment by tknorris** [comment]
No issues here.

**Comment by Razzeee** [comment]
Thanks, I'm going to push it live then

**Comment by Razzeee** [comment]
Can you push a new version of the dev branch + https://github.com/Razzeee/script.module.arrow and https://github.com/Razzeee/script.module.trakt

**Comment by tknorris** [comment]
yep, will do.

**Comment by RavRob** [comment]
Arrow and Trakt.py both fail to auto-update.

**Comment by RavRob** [comment]
Oh and Trakt also fails to auto-update.  So all 3 are failing.

**Comment by tknorris** [comment]
github frequently rejects xbmc http requests after update. wait a bit or uninstall/install trakt and it'll install

**Comment by RavRob** [comment]
It installed fine on Windows 8.1 with Isenguard 15.1.  It was failling on
Windows 10 running Jarvis Alpha 2.  After many times rebooting and
re-trying, it finally went through.

Tx

**Comment by Razzeee** [comment]
Anyone having problems?

**Comment by RavRob** [comment]
Nothing to report at my end.  Still working like clock work.

**Comment by Razzeee** [comment]
I would like some people to test this branch, but please please don't use your normal user as things might be broken and trakt data might be written wrong. Will only work with JARVIS or above.
https://github.com/Razzeee/script.trakt/tree/User-Ratings

This should sync movie ratings when syncing. (as in download them to trakt) 
And when rating something it should also be set in trakt as a rating.

**Comment by s7eele** [comment]
Heard. Will do what I can to assist:)

**Comment by tknorris** [comment]
I can push this to my beta repo, but quite a few people are using it these day (unfortunately). What's the worst that can happen? It corrupts your trakt ratings?

**Comment by Razzeee** [comment]
I would have asked for that, if I wanted that :)
I needed to touch the movie compare method, so I would like to be safe. Changes should really be fine, but one never knows.

https://github.com/Razzeee/script.trakt/commit/5b0b56d32d2400630e70f543b7761c3bb71c626f#diff-b7d4b8dcb174f921d1e08b9484b1b6a4R859

**Comment by tknorris** [comment]
I thought that's what you were asking for? Are you asking for some people to install and test from github?

**Comment by Razzeee** [comment]
Yes, some manual installs / tests would be great this time.

**Comment by Razzeee** [comment]
So has anyone tested anything? I've got one tester commenting that everything seems fine with that change.

**Comment by tknorris** [comment]
I haven't been able to test it. I don't use Kodi ratings.

**Comment by Razzeee** [comment]
Well, I'm more interested if it has regressions on the collection/watched side of things

**Comment by s7eele** [comment]
Sorry, had something I had to take care of, will try to test through tonight.

**Comment by Razzeee** [comment]
I just added episode and tvshow ratings sync. So this is now feature complete.

**Comment by s7eele** [comment]
@Razzeee, I haven't been able to get to this, busy with previous commitment that I haven't finished, it is actually trakt related as well so can't do simultaneously, still needed?

**Comment by Razzeee** [comment]
Well would be nice, but it feels like it might be ready for prime time already

**Comment by RavRob** [comment]
I have had no issues with it.  But likely not tested as thoroughly as I
should have.

**Comment by s7eele** [comment]
Should have kept my mouth shut, things just keep getting worse:), car died this morning, won't be able to get to it anytime real soon:/

**Comment by Razzeee** [comment]
Had to make some changes, would be appreciated if you can test again

**Comment by Razzeee** [comment]
@tknorris did some more tests today and it seems to be fail safe, can you push the branch to your repo?

**Comment by tknorris** [comment]
Yep, sure will. Is the ratings sync on or off by default?

**Comment by Razzeee** [comment]
it's on by default

**Comment by tknorris** [comment]
3.1.0~alpha1 pushed including everything thru this commit: https://github.com/Razzeee/script.trakt/commit/caa390155ca2d10c150b92324b21fd6953cbddde

**Comment by Razzeee** [comment]
Thank you

**Comment by tknorris** [comment]
FYI, things are screwed up right now because of the .org repo being down. When it tried to update on my system it got marked as broken.

**Comment by Razzeee** [comment]
Yes, 3.1.0 needs Jarvis at least. 
I actually do this in preparation for the Jarvis final.

**Comment by Razzeee** [comment]
Then forget about the beta repo as you probably can't target versions

**Comment by tknorris** [comment]
I'd have to push a new repo zip out. 3.1.0 won't work on Isengard at all? I guess I can roll back the change, I'm not sure what Kodi will do in that case.

**Comment by Razzeee** [comment]
No, 3.1.0 uses Jarvis specific features (ratings) 
So it's not backwards compatible.

**Comment by tknorris** [comment]
I'll revert the commit for now, hopefully that'll clear it from everyone's update queue. When you push this to .org repo I guess you need to make sure to only push it to the Jarvis one.

**Comment by Razzeee** [comment]
Yes, that's the plan. I forgot about that for the beta repo unfortunately.

**Comment by tknorris** [comment]
No problem, I rolled back the commit and it seemed to rollback the update on kodi. It's too bad we can't just disable the ratings function for pre-Jarvis...

**Comment by Razzeee** [comment]
Well that's are no other features in that patch other than some cleaning and maybe accidental Bugfixes. 
So as far as I am considered everything prior is end of life unless there comes up a major problem.
